pkg/kernel/builder.go

```diff
@@ -0,0 +1,32 @@
+package kernel
+
+import (
+	"context"
+
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+func BuildFactory(ctx context.Context, config cfg.Config, logger log.Logger, options []Option) (*factory, error) {
+	blueprint := NewBlueprint(options...)
+
+	var err error
+	var factory *factory
+
+	if factory, err = NewFactory(ctx, config, logger, blueprint); err != nil {
```

Comment:
Is `NewFactory` returning a non-nil factory if it returns an error? If not, we can directly return the result from it, right?